packages/trader/src/AppV2/Components/OrderDetails/order-details.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { Text } from '@deriv-com/quill-ui';
+import useOrderDetails from 'AppV2/Hooks/useOrderDetails';
+import { TContractInfo } from '@deriv/shared';
+import CardWrapper from 'AppV2/Components/CardWrapper';
+import { Localize } from '@deriv/translations';
+
+interface ContractInfoProps {
+    contract_info: TContractInfo;
+}
+
+const OrderDetails = ({ contract_info }: ContractInfoProps) => {
+    const orderDetails = useOrderDetails(contract_info);
+    const details = orderDetails ? orderDetails.details : {};
+    return (
+        <CardWrapper title={<Localize i18n_default_text='Order Details' />}>
+            <div className='order-details'>
+                <div className='order-details__table'>
+                    {Object.entries(details).map(([key, value]) => (
+                        <div className='order-details__table-row' key={key}>
+                            <div className='order-details__table-row-cell'>
+                                <Text size='sm' color='quill-typography__color--subtle'>
+                                    {key}
+                                </Text>
+                            </div>
+                            <div className='order-details__table-row-cell'>
+                                {Array.isArray(value) ? (
+                                    value.map(val => (
+                                        <Text key={val} size='sm'>
+                                            {val}
+                                        </Text>
+                                    ))
+                                ) : (
+                                    <Text size='sm'>{value as string | number}</Text>
```

Comment:
This type assertion is needed because TS doesn't recognize `Array.isArray` check? And if we leave just `value as string`  - should it be enough (without number)?

Reply:
I think this is not needed here, I'll just remove it